pinot-core/src/main/java/com/linkedin/pinot/core/io/readerwriter/impl/FixedByteSingleColumnMultiValueReaderWriter.java

```diff
@@ -234,9 +282,11 @@ public void setBytesArray(int row, byte[][] bytesArray) {
 
   @Override
   public int getCharArray(int row, char[] charArray) {
-    int bufferIndex = headerReader.getInt(row, 1);
-    int startIndex = headerReader.getInt(row, 1);
-    int length = headerReader.getInt(row, 2);
+    FixedByteSingleValueMultiColReader headerReader = getCurrentReader(row);
```

Comment:
[again speaking aloud] we should be able to bulk copy....
Another point is, for our existing data, most of the time we write one value (inputArray.length == 1). We should special case that with if-else because setting loops is costlier. Not sure how jvm optimizes that at runtime. Needs benchmarking